### PR TITLE
Apply CheckDeps, InitDeps, and CloseDeps to existing plugin

### PR DESCRIPTION
**Pull request by edwarnicke**
Added CheckDeps to options_tests.go files for plugins

Use InitDeps() and CloseDeps() rather than explicit
Init and close of Deps in Plugins.

Note: this also allows us to switch the interface for Deps from
plugin.PluginAPI to plugin.API in most places.

### CODE REVIEW

File: plugins/crd/plugin_impl_crd.go [hunk 1]
```diff
@@ -77,8 +79,8 @@ type Deps struct {
 	Log  logger.FieldLoggerPlugin
 	// Kubeconfig with k8s cluster address and access credentials to use.
 	KubeConfig  string
-	Handler     handler.PluginAPI
-	ObjectStore objectstore.PluginAPI
+	Handler     handler.API
```
Comment (sbezverk) [inline plugins/crd/plugin_impl_crd.go]: Handler plugin has not been used in CRD plugin, if you check the code it is only defined here and used in tests, no other references. I would suggest to remove it from CRD plugin completely.

Comment (edwarnicke) [reply plugins/crd/plugin_impl_crd.go]: @sbezverk Happy too.   Can we do it in a separate PR?

**Comment by sbezverk** [comment]
@edwarnicke yep, separate PR work for me. Let's just not forget about it.

**Comment by fkautz** [comment]
This patch should block until #170 is merged.

**Pull request merged**